controllers/syncer/call_home.go

```diff
@@ -0,0 +1,168 @@
+	obj := &batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        config.GetNameForResource(config.CallHome, driver.Name),
+			Namespace:   driver.Namespace,
+			Annotations: driver.GetAnnotations("", ""),
+			Labels:      driver.GetLabels(),
+		},
+	}
+
+	sync := &callHomeSyncer{
+		driver: driver,
+		obj:    obj,
+	}
+
+	return syncer.NewObjectSyncer(config.CallHome.String(), driver.Unwrap(), obj, c, func() error {
+		return sync.SyncFn()
+	})
+}
+
+func (s *callHomeSyncer) SyncFn() error {
+	out := s.obj.(*batchv1.CronJob)
+
+	out.Spec.Schedule = CronSchedule
+
+	// ensure template
+	out.Spec.JobTemplate.ObjectMeta.Labels = s.driver.GetCallHomePodLabels()
+	out.Spec.JobTemplate.ObjectMeta.Annotations = s.driver.GetAnnotations("", "")
+	out.Spec.JobTemplate.Spec.Template.ObjectMeta.Labels = s.driver.GetCallHomePodLabels()
+	successfulJobsHistoryLimit := jobsHistoryLimit
+	out.Spec.SuccessfulJobsHistoryLimit = &successfulJobsHistoryLimit
+
+	err := mergo.Merge(&out.Spec.JobTemplate.Spec.Template.Spec, s.ensurePodSpec(), mergo.WithTransformers(transformers.PodSpec))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *callHomeSyncer) ensurePodSpec() corev1.PodSpec {
+	return corev1.PodSpec{
+		Containers:         s.ensureContainersSpec(),
+		Volumes:            s.ensureVolumes(),
+		ServiceAccountName: config.GetNameForResource(config.CallHomeServiceAccount, s.driver.Name),
+		Affinity:           s.driver.Spec.CallHome.Affinity,
+		Tolerations:        s.driver.Spec.CallHome.Tolerations,
+		RestartPolicy:      "OnFailure",
+	}
+}
+
+func (s *callHomeSyncer) ensureContainersSpec() []corev1.Container {
+	callHomePlugin := s.ensureContainer(callHomeContainerName,
+		s.driver.GetCallHomeImage(),
+		[]string{"--csi-endpoint=$(CSI_ENDPOINT)"},
+	)
+
+	callHomePlugin.Resources = ensureResources("40m", "800m", "40Mi", "400Mi")
+
+	healthPort := s.driver.Spec.HealthPort
+	if healthPort == 0 {
+		healthPort = controllerContainerDefaultHealthPortNumber
+	}
+
+	callHomePlugin.Ports = ensurePorts(corev1.ContainerPort{
+		Name:          controllerContainerHealthPortName,
+		ContainerPort: int32(healthPort),
+	})
+	callHomePlugin.ImagePullPolicy = s.driver.Spec.CallHome.ImagePullPolicy
+
+	return []corev1.Container{
+		callHomePlugin,
+	}
+}
+
+func (s *callHomeSyncer) ensureContainer(name, image string, args []string) corev1.Container {
+	sc := &corev1.SecurityContext{AllowPrivilegeEscalation: boolptr.False()}
+	fillSecurityContextCapabilities(sc)
+	return corev1.Container{
+		Name:            name,
+		Image:           image,
+		Args:            args,
+		Env:             s.getEnv(),
+		VolumeMounts:    s.getVolumeMounts(),
+		SecurityContext: sc,
+		Resources:       ensureDefaultResources(),
+	}
+}
+
+func (s *callHomeSyncer) getEnv() []corev1.EnvVar {
+
+	return []corev1.EnvVar{
+		{
+			Name:  "CSI_ENDPOINT",
+			Value: config.CSIEndpoint,
```

Comment:
?

Reply:
done

Reply:
because we are planning on using the `CSI_LOGLEVEL` env var?

Reply:
ok, removed